### PR TITLE
peer: abstractions to aid in unit testing

### PR DESCRIPTION
**Pull request by Crypt-iQ**
Commit:
- `MessageConn` interface that gets rid of the dependency on `*brontide.Conn` & remove `Config` from config.go per @halseth

### CODE REVIEW

File: peer/brontide.go [hunk 1]
```diff
@@ -693,7 +692,7 @@ func (p *Brontide) String() string {
 // readNextMessage reads, and returns the next message on the wire along with
 // any additional raw payload.
 func (p *Brontide) readNextMessage() (lnwire.Message, error) {
-	noiseConn, ok := p.cfg.Conn.(*brontide.Conn)
+	noiseConn, ok := p.cfg.Conn.(MessageConn)
```
Comment (halseth) [inline peer/brontide.go:907]: could we change the type of `p.cfg.Conn` to `MessageConn`?

**Review by halseth — APPROVED** [review]
LGTM ✅

File: peer/brontide.go (hunk 1 above)
Comment (Crypt-iQ) [reply peer/brontide.go:907]: Yeah that makes sense

File: server.go [hunk 2]
```diff
@@ -2946,7 +2946,7 @@ func (s *server) peerConnected(conn net.Conn, connReq *connmgr.ConnReq,
 	// htlcs, an extra block is added to prevent the channel from being
 	// closed when the htlc is outstanding and a new block comes in.
 	pCfg := peer.Config{
-		Conn:                    conn,
+		Conn:                    conn.(*brontide.Conn),
```
Comment (halseth) [inline server.go:2949]: can just use `Conn: brontideConn`

Comment (Crypt-iQ) [reply server.go]: There are two other places `*brontide.Conn` is used but I think for now it's ok to leave them and have a follow-up to tackle them.

**Review by halseth — APPROVED** [review]
LGTM pending squash 🌡️

**Review by bhandras — APPROVED** [review]
LGTM :+1:

**Pull request merged**